llvm/include/llvm/Transforms/Utils/Cloning.h

```diff
@@ -205,6 +205,11 @@ void CloneAndPruneFunctionInto(Function *NewFunc, const Function *OldFunc,
                                const char *NameSuffix = "",
                                ClonedCodeInfo *CodeInfo = nullptr);
 
+/// Process debug information from function's subprogram attachment.
+DISubprogram *ProcessSubprogramAttachment(const Function &F,
```

Comment:
Could you elaborate on "Process"? This is more about collecting debug info, right?

Reply:
That's right. This function hydrates the passed DebugInfoFinder by visiting relevant components of the function's subprogram attachment. I'll update the name/comment when we get to extracting that commit in a separate PR.

Reply:
Sounds good, now that the other two PRs are open, let's proceed with this commit!

Reply:
Great, yes! @felipepiovezan a silly question -- now that those PRs are approved, how/when do they get merged? Stacking unmerged PRs on top of each other doesn't have the best UX in Github.

Reply:
So I think the best option here is to merge the two PRs that are approved (https://github.com/llvm/llvm-project/pull/112976 and https://github.com/llvm/llvm-project/pull/112948) . Both of those stand on their own and don't depend on each other, so it's fine to merge in w/e order, right?

And then you can grab the third commit (or more, if the subsequent commits are also independent of each other) of _this_ PR and open a new PR (or multiple PRs, if you did the () before), so it will look fine on top of `main`. My understanding is that the fourth commit depends on the third though.

Regarding what to do about this umbrella PR in the mean time, you have two options: leave this as is (with the understanding that it is not meant to be merged, merely serve as a reference for all the work left to do), or rebase on top of main, which will get rid of the two-already merged commits and force push to your branch `artempyanykh:fast-coro-upstream`, which will mean this PR serves a reference of the work that's left to be merged, also with the understand that we're not merging it.

With all that stack, we'd never have to stack unmerged PRs on top of each other, does that make sense?

Reply:
By the way, do you have commit access?

Reply:
> So I think the best option here is to merge the two PRs that are approved (https://github.com/llvm/llvm-project/pull/112976 and https://github.com/llvm/llvm-project/pull/112948) . Both of those stand on their own and don't depend on each other, so it's fine to merge in w/e order, right?

That's right, yes.

> And then you can grab the third commit (or more, if the subsequent commits are also independent of each other) of this PR and open a new PR (or multiple PRs, if you did the () before), so it will look fine on top of main. My understanding is that the fourth commit depends on the third though.

After the first 2 are merged, extracting the third one in a separate PR is straightforward. With the rest of the stack there's quite a bit of sequencing, so we may need to rinse and repeat.

> rebase on top of main, which will get rid of the two-already merged commits and force push to your branch artempyanykh:fast-coro-upstream, which will mean this PR serves a reference of the work that's left to be merged, also with the understand that we're not merging it

+1 to this. I was planning to keep rebasing this stack anyway to make sure things keep working end-to-end while the chunks are being extracted and merged.

> By the way, do you have commit access?

No, not sure what the process for this is. Why?

Reply:
Sorry for the delay, the LLVM dev conference happened last week and we were mostly out.

> No, not sure what the process for this is. Why?

Just wondering if you needed help merging the open PRs. I've merged the first two for you.
The process for getting commit access is described here: https://llvm.org/docs/DeveloperPolicy.html#obtaining-commit-access
If you don't want to do that, you can always let the reviewers know you need someone to press the merge button for you

Reply:
> Just wondering if you needed help merging the open PRs. I've merged the first two for you.

Thank you @felipepiovezan, appreciate the help! I also emailed Chris and got the commit rights.

I'll go ahead and extract the next PR.